src/AcquisitionService.php

```diff
@@ -171,6 +179,13 @@ protected function findMatch(array $values, array &$context) {
       $query->exists('name');
     }
 
+    // If we have the don't have the include protected roles behaviour, ensure
+    // protected roles are excluded.
+    if ($context['behavior'] ^ self::BEHAVIOR_INCLUDE_PROTECTED_ROLES) {
+      $roles = $this->settings->get('acquisitions.protected_roles');
+      $query->condition('roles', $roles, 'NOT IN');
```

Comment:
We will need to check that we don't have no roles (empty array), as that will break the query.